tinygrad/codegen/kernel.py

```diff
@@ -511,7 +511,8 @@ def has_expanded_axis(shape, strides): return any(s > 1 and st == 0 for s,st in
         for sz in ([256, 16] if prod(self.sts[0].shape[:self.first_reduce]) <= 32 else [16]):
           if all(st.shape[self.first_reduce] % sz == 0 or st.shape[self.first_reduce] == 1 for st in self.sts):
             try: # may fail due to excessive smem usage
-              self.apply_opt(Opt(OptOps.GROUPTOP, 0, sz))
+              if len(self.reduceops) < 2:
```

Comment:
but nothing breaks?
![image](https://github.com/user-attachments/assets/522b2516-dea7-4743-be8d-7db96275cc47)

Reply:
I don't understand, you pushed something that removed it, why did it exist in the first place?

Reply:
I started working on this earlier; it was meant to prevent anything from sneaking through `hand_coded_optimizations` while I was still testing.

I meant to delete it, the cases that were causing trouble are covered by `test_partial_opt_multireduce`